src/Filesystem/Replicate.php

```diff
@@ -193,17 +198,28 @@ public function getAdapterClassNames()
 
     public function createFile($key, Filesystem $filesystem)
     {
-        return $this->primary->createFile($key, $filesystem);
-    }
+        if ($this->primary instanceof FileFactory) {
+            return $this->primary->createFile($key, $filesystem);
+        }
 
-    public function createFileStream($key, Filesystem $filesystem)
-    {
-        return $this->primary->createFileStream($key, $filesystem);
+        if ($this->secondary instanceof FileFactory) {
+            return $this->secondary->createFile($key, $filesystem);
+        }
+
+        throw new \RuntimeException(sprintf('None of the adapters implements: %s', FileFactory::class));
```

Comment:
```suggestion
        throw new \LogicException(sprintf('None of the adapters implements %s.', FileFactory::class));
```
Even, I'd say that this exception could be of type `\TypeError`, since the invalid flow is caused by a value not respecting the expected type.